### ISSUE TITLE
There is a bug in main: self.loggedIn : 113 - leading to Cannot read property 'reminders' of undefined

### ISSUE THREAD

**Issue by paulroth3d** [MauriceConrad/iCloud-API#83]
TLDR; @MauriceConrad  - can you please release a new version - as there is a bug on main.js:113 that is fixed in the repo - BUT NOT if anyone `npm installs..`

---

There is a bug in main.js:113

it seems it is FIXED in the repository, BUT NOT IN npm.

This is what you get on line 113 if you install `apple-icloud`

`npm install --save apple-icloud`

look at `node_modules/apple-icloud/main.js`, and you get this:

```
      self.loggedIn = (function() {
        //console.log(self.auth.cookies.length > 0, !!self.auth.token, self.account != {}, self.username === username);
        return (self.auth.cookies.length > 0 && self.auth.token && self.account != {} && username ? (self.username === username) : true);
      })();
```

note the final condition: `&& username ? (self.username === username) : true`

this is wrong and needs to be: `&& (username ? (self.username === username) : false));`

(missing the parentheses around the check - bypassing the && operator, and almost always returning true)

*This appears fixed in the git repository*

https://github.com/MauriceConrad/iCloud-API/blob/master/main.js#L115

---

This is actually the source of a NUMBER of issues:

https://github.com/MauriceConrad/iCloud-API/issues/64
https://github.com/MauriceConrad/iCloud-API/issues/71
https://github.com/MauriceConrad/iCloud-API/issues/53
https://github.com/MauriceConrad/iCloud-API/issues/47

and many others.

These - and many others - all come from `myCloud.account` being an empty object, and therefore none of the apis can be called.

The reason it is empty is because `self.loggedIn` is setting to true, and therefore the `self.login` on line 162 never fires.

Could you please release a new version to npm?

**Comment by paulroth3d** [comment]
Otherwise, for anyone else running into this,
look in the `node_modules/apple-icloud/main.js`

and change this:

```
      // Now, validate the session with checking for important aspects that show that the session can be used to get data (e.g. there need to be a token, some cookies and account info)
      self.loggedIn = (function() {
        //console.log(self.auth.cookies.length > 0, !!self.auth.token, self.account != {}, self.username === username);
        return (self.auth.cookies.length > 0 && self.auth.token && self.account != {} && username ? (self.username === username) : true);
      })();
```

into this:

```
      // Now, validate the session with checking for important aspects that show that the session can be used to get data (e.g. there need to be a token, some cookies and account info)
      self.loggedIn = (function() {
        //console.log(self.auth.cookies.length > 0, !!self.auth.token, self.account != {}, self.username === username);
        return (self.auth.cookies.length > 0 && self.auth.token && self.account != {} && username ? ((self.username === username) : false));
      })();
```

**Comment by wildesjamer** [comment]
thank you for the fix, but i get a error - maybe you can help me with that

```
SyntaxError: Unexpected token '<', "<html>
<h"... is not valid JSON
    at JSON.parse (<anonymous>)
    at Request._callback (D:\dev\node-Projects\webuntis\node_modules\apple-icloud\setup.js:44:29)
    at self.callback (D:\dev\node-Projects\webuntis\node_modules\request\request.js:185:22)
    at Request.emit (node:events:519:28)
    at Request.<anonymous> (D:\dev\node-Projects\webuntis\node_modules\request\request.js:1154:10)
    at Request.emit (node:events:519:28)
    at IncomingMessage.<anonymous> (D:\dev\node-Projects\webuntis\node_modules\request\request.js:1076:12)
    at Object.onceWrapper (node:events:633:28)
    at IncomingMessage.emit (node:events:531:35)
    at endReadableNT (node:internal/streams/readable:1696:12)
```

**Comment by wildesjamer** [comment]
now i updated the setup.js too and have a new error

```
{ error: 'Session is expired or invalid', errorCode: 6 }
{
  action: 'start',
  parentAction: 'login',
  progress: 0,
  message: 'Trying to reset session and login'
}
{
  error: 'Failed to parse token.',
  code: 0,
  requestBody: '<html>\r\n' +
    '<head><title>503 Service Temporarily Unavailable</title></head>\r\n' +
    '<body>\r\n' +
    '<center><h1>503 Service Temporarily Unavailable</h1></center>\r\n' +
    '<hr><center>Apple</center>\r\n' +
    '</body>\r\n' +
    '</html>\r\n'
}
```
